Runtimes/Core/CMakeLists.txt

```diff
@@ -143,7 +143,6 @@ add_compile_definitions(
   $<$<BOOL:${SwiftCore_ENABLE_OBJC_INTEROP}>:-DSWIFT_OBJC_INTEROP>
   $<$<BOOL:${SwiftCore_ENABLE_LIBRARY_EVOLUTION}>:-DSWIFT_LIBRARY_EVOLUTION>
   $<$<BOOL:${SwiftCore_ENABLE_CRASH_REPORTER_CLIENT}>:-DSWIFT_HAVE_CRASHREPORTERCLIENT>
-  $<$<BOOL:${SwiftCore_ENABLE_VECTOR_TYPES}>:-DSWIFT_STDLIB_ENABLE_VECTOR_TYPES>
```

Comment:
To double check my understanding: are we pushing this setting to `runtime` and `core` because those are the only artifacts for which it makes sense?

Reply:
I moved it to the public interface of `runtime` so that it affects `runtime` and anything that links against it. The issue is that `runtime` is privately linked into `core`, so the macro won't get carried out to products that link against `swiftCore` externally.

Then I made it part of the `INTERFACE` of `swiftCore`. It doesn't get re-applied to `swiftCore`, but is forwarded to anything that links against `swiftCore` using the config file. This means that the overlays and supplemental libraries that are using this config file will also get the flag set without needing to do anything.